pallets/asset-manager/src/migrations.rs

```diff
@@ -0,0 +1,416 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::{AssetIdType, AssetTypeId, AssetTypeUnitsPerSecond, Config};
+use frame_support::{
+	pallet_prelude::PhantomData,
+	storage::migration::{remove_storage_prefix, storage_key_iter},
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+	Blake2_128Concat,
+};
+use sp_std::convert::TryInto;
+//TODO sometimes this is unused, sometimes its necessary
+use sp_std::vec::Vec;
+use xcm::latest::prelude::*;
+
+pub struct UnitsWithAssetType<T>(PhantomData<T>);
```

Comment:
Some documentation on what these do would be good here (or at least a pointer to where it's documented....?)